CriticalMass/RequestManager.swift

```diff
@@ -120,3 +124,25 @@ public class RequestManager {
         }
     }
 }
+
+@available(iOS 12.0, *)
+private extension RequestManager {
+    func setup(pathMonitor: NWPathMonitor) {
+        pathMonitor.pathUpdateHandler = { [weak self] path in
+            self?.handlePathUpdate(path)
+        }
+        let queue = DispatchQueue(label: "Monitor")
+        pathMonitor.start(queue: queue)
+        
+        pathMonitorRef = pathMonitor
+    }
+
+    func handlePathUpdate(_ path: NWPath) {
+        if path.status == .satisfied {
```

Comment:
Isn't this also get called when the network changes from Wifi to cellular?

Reply:
I guess it does get called. When I implement adapter for NWPathMonitor, I will track only status changes

Reply:
By the way I can refactor LocationManager into operation, not a problem